docs/standard/io/how-to-write-text-to-a-file.md

```diff
@@ -14,79 +14,55 @@ ms.assetid: 060cbe06-2adf-4337-9e7b-961a5c840208
+- <xref:System.IO.Path> is for strings that have file or directory path information. It contains the <xref:System.IO.Path.Combine%2A> method and, in .NET Core 2.1 and later, the <xref:System.IO.Path.Join%2A> and <xref:System.IO.Path.TryJoin%2A> methods, which allow concatenation of strings to build a file or directory path.
 
-- <xref:System.IO.Path> - to be used on strings that contain file or directory path information. It contains the <xref:System.IO.Path.Combine%2A> method, which allows concatenation of strings to build a file or directory path.
+> [!NOTE]
+> The following examples show only the minimum amount of code needed. A real-world app usually provides more robust error checking and exception handling.  
+  
+## Example: Synchronously write text with StreamWriter
 
+The following examples shows how to synchronously write text to a new file using the <xref:System.IO.StreamWriter> class, one line at a time. Because the <xref:System.IO.StreamWriter> object is declared and instantiated in a `using` statement, the <xref:System.IO.StreamWriter.Dispose%2A> method is invoked, which automatically flushes and closes the stream.  
+
+[!code-csharp[Conceptual.BasicIO.TextFiles#WriteLine](../../../samples/snippets/csharp/VS_Snippets_CLR/conceptual.basicio.textfiles/cs/write.cs)] 
+[!code-vb[Conceptual.BasicIO.TextFiles#WriteLine](../../../samples/snippets/visualbasic/VS_Snippets_CLR/conceptual.basicio.textfiles/vb/write.vb)]  
+
+## Example: Synchronously append text with StreamWriter
+
+The following example shows how to synchronously append text to the text file created in the first example, using the <xref:System.IO.StreamWriter> class.   
+
+[!code-csharp[Conceptual.BasicIO.TextFiles#WriteLine](../../../samples/snippets/csharp/VS_Snippets_CLR/conceptual.basicio.textfiles/cs/append.cs)] 
+[!code-vb[Conceptual.BasicIO.TextFiles#WriteLine](../../../samples/snippets/visualbasic/VS_Snippets_CLR/conceptual.basicio.textfiles/vb/append.vb)]  
+
+## Example: Asynchronously write text with StreamWriter
+
+The following example shows how to asynchronously write text to a new file using the <xref:System.IO.StreamWriter> class. To invoke the <xref:System.IO.StreamWriter.WriteAsync%2A> method, the method call must be within an `async` method. This example requires C# 7.1 or later, which adds support for the `async` modifier on the program entry point. 
+
+[!code-csharp[Conceptual.BasicIO.TextFiles#WriteLine](../../../samples/snippets/csharp/VS_Snippets_CLR/conceptual.basicio.textfiles/cs/async.cs)] 
+[!code-vb[Conceptual.BasicIO.TextFiles#WriteLine](../../../samples/snippets/visualbasic/VS_Snippets_CLR/conceptual.basicio.textfiles/vb/async.vb)]  
```

Comment:
The VB code doesn't compile, since Async Main isn't supported. The code has to be:
```
Imports System.IO

Public Module Example
   Public Sub Main()
      WriteTextAsync()
   End Sub

   Async Sub WriteTextAsync() 
       ' Set a variable to the Documents path.
       Dim docPath As String = Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments)

       ' Write the text asynchronously to a new file named "WriteTextAsync.txt".
        Using outputFile As New StreamWriter(Path.Combine(docPath, Convert.ToString("WriteTextAsync.txt")))
           Await outputFile.WriteAsync("This is a sentence.")
       End Using
   End Sub
End Module
```